.github/workflows/reusable-upgrade-latest-test.yaml

```diff
@@ -0,0 +1,92 @@
+name: Reusable workflow - upgrade latest with cli test
+
+on:
+  workflow_call:
+    inputs:
+      flavor:
+        required: true
+        type: string
+
+jobs:
+  # This job is used to download the latest release and then used by the other job
+  latest-release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: robinraju/release-downloader@v1.8
+        with:
+          # A flag to set the download target as latest release
+          # The default value is 'false'
+          latest: true
+          repository: kairos-io/kairos
+          fileName: '*opensuse-leap-v*.iso'
+          out-file-path: last-release
+      - uses: actions/upload-artifact@v3
+        with:
+          name: latest-release.zip
+          path: last-release
+          if-no-files-found: error
+
+  build:
+    needs:
+    - latest-release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          git fetch --prune --unshallow
+      - name: Download latest release
+        uses: actions/download-artifact@v3
+        with:
+          name: latest-release.zip
+      - name: Download iso
+        uses: actions/download-artifact@v3
+        with:
+          name: kairos-${{ inputs.flavor }}.iso.zip
```

Comment:
do we need to unpack this? see line 83 ... before we didn't need to because it was all in the same job